ui/xui/gl-helpers.cc

```diff
@@ -774,3 +774,25 @@ void SaveScreenshot(GLuint tex, bool flip)
         free(msg);
     }
 }
+
+void export_texture(ExportData *imagedata)
+{
+    char filename[13];
+    std::vector<uint8_t> pixels;
+    std::vector<uint8_t> png;
+    
+    snprintf(filename, sizeof(filename), "%lx", imagedata->vram_addr);
+    FILE *texture = fopen(filename, "wb");
+  
+    pixels.resize(imagedata->clip_width * imagedata->clip_height * imagedata->surface_scale_factor * imagedata->bytes_per_pixel);
+   glReadPixels(imagedata->clip_x, imagedata->clip_y, imagedata->clip_width, imagedata->clip_height, 
```

Comment:
This is being called from the UI thread and I don't think there's any guarantee that the pthread thread is stopped while this happens. This means that even in the best case scenario the texture binding has probably changed by the time this is called so you will read pixels from some other texture than the one you want. You would be much better off saving the pixels on the pgraph side with just the triggering logic on the UI thread.